src/init.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import imageLoader from './utils/imageLoader';
+import type { Config } from './simpleImgProvider';
+
+export default function init(config: Config) {
+  /* eslint-disable */
+  if (!window.IntersectionObserver) require('intersection-observer');
+  // $FlowIgnoreLine:
+  const observer = new IntersectionObserver(entries => onIntersection.call(this, entries), config);
+  return this ? observer : window.reactSimpleImgObserver = observer;
+  /* eslint-enable */
+}
+
+export function onIntersection(entries: Array<{ intersectionRatio: number, target: any }>) {
+  entries.forEach(({ intersectionRatio, target }) => (intersectionRatio > 0) && imageLoader.call(this, target));
```

Comment:
You need to break out of the for loop (you can't break with forEach so you'll have to change this to a plain for loop or use find) once you detect `intersectionRation > 0`.